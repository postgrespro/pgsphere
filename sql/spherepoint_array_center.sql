SELECT center(ARRAY[
    spoint(40.7128, -74.0060),
    spoint(34.0522, -118.2437),
    spoint(37.7749, -122.4194)
]);

SELECT center('{}'::SPoint[]);

CREATE FUNCTION spoint_from_xyz(FLOAT8, FLOAT8, FLOAT8)
   RETURNS spoint
   AS 'pg_sphere', 'spoint_from_xyz'
   LANGUAGE 'c'
   IMMUTABLE STRICT PARALLEL SAFE;

SELECT spoint_from_xyz(1, 0, 0);

SELECT spoint_from_xyz(0, 0, 0);

SELECT center(ARRAY[
    spoint_from_xyz(1, 0, 0),
    spoint_from_xyz(-1, 0, 0)
]);

SELECT center(NULL::SPoint[]);

SELECT @@ ARRAY[
    spoint(40.7128, -74.0060),
    spoint(34.0522, -118.2437),
    spoint(37.7749, -122.4194)
] AS center;

SELECT @@ ARRAY[]::spoint[] AS center;

SELECT @@ ARRAY[
    spoint_from_xyz(1, 0, 0),
    spoint_from_xyz(-1, 0, 0)
] AS center;
