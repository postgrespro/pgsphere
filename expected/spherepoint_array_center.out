SELECT center(ARRAY[
    spoint(40.7128, -74.0060),
    spoint(34.0522, -118.2437),
    spoint(37.7749, -122.4194)
]);
                 center                 
----------------------------------------
 (3.04366980631979 , 0.858938068921891)
(1 row)

SELECT center('{}'::SPoint[]);
 center 
--------
 
(1 row)

CREATE FUNCTION spoint_from_xyz(FLOAT8, FLOAT8, FLOAT8)
   RETURNS spoint
   AS 'pg_sphere', 'spoint_from_xyz'
   LANGUAGE 'c'
   IMMUTABLE STRICT PARALLEL SAFE;
SELECT spoint_from_xyz(1, 0, 0);
 spoint_from_xyz 
-----------------
 (0 , 0)
(1 row)

SELECT spoint_from_xyz(0, 0, 0);
 spoint_from_xyz 
-----------------
 (0 , 0)
(1 row)

SELECT center(ARRAY[
    spoint_from_xyz(1, 0, 0),
    spoint_from_xyz(-1, 0, 0)
]);
        center         
-----------------------
 (1.5707963267949 , 0)
(1 row)

SELECT center(NULL::SPoint[]);
 center 
--------
 
(1 row)

SELECT @@ ARRAY[
    spoint(40.7128, -74.0060),
    spoint(34.0522, -118.2437),
    spoint(37.7749, -122.4194)
] AS center;
                 center                 
----------------------------------------
 (3.04366980631979 , 0.858938068921891)
(1 row)

SELECT @@ ARRAY[]::spoint[] AS center;
 center 
--------
 
(1 row)

SELECT @@ ARRAY[
    spoint_from_xyz(1, 0, 0),
    spoint_from_xyz(-1, 0, 0)
] AS center;
        center         
-----------------------
 (1.5707963267949 , 0)
(1 row)

